x-pack/plugins/stack_alerts/public/alert_types/es_query/expression/search_source_expression_form.tsx

```diff
@@ -0,0 +1,265 @@
+          withDebounce(() => setParam(action.type, action.payload));
+        }
+        return { ...currentState, [action.type]: action.payload };
+      },
+      {
+        index: searchSource.getField('index')!,
+        query: searchSource.getField('query')!,
+        filter: mapAndFlattenFilters(searchSource.getField('filter') as Filter[]),
+        threshold: ruleParams.threshold,
+        timeWindowSize: ruleParams.timeWindowSize,
+        size: ruleParams.size,
+      }
+    );
+  const dataViews = useMemo(() => [dataView], [dataView]);
+
+  const onSelectDataView = useCallback(
+    (newDataViewId) =>
+      data.dataViews
+        .get(newDataViewId)
+        .then((newDataView) => dispatch({ type: 'index', payload: newDataView })),
+    [data.dataViews]
+  );
+
+  const onUpdateFilters = useCallback((newFilters) => {
+    dispatch({ type: 'filter', payload: mapAndFlattenFilters(newFilters) });
+  }, []);
+
+  const onChangeQuery = useCallback(
+    ({ query: newQuery }: { query?: Query }) => {
+      withDebounce(() => dispatch({ type: 'query', payload: newQuery || { ...query, query: '' } }));
+    },
+    [query]
+  );
+
+  const onSavedQueryUpdated = useCallback((newSavedQuery: SavedQuery) => {
+    setSavedQuery(newSavedQuery);
+    const newFilters = newSavedQuery.attributes.filters;
+    if (newFilters) {
+      dispatch({ type: 'filter', payload: newFilters });
+    }
+  }, []);
+
+  const onClearSavedQuery = () => {
+    setSavedQuery(undefined);
+    dispatch({ type: 'query', payload: { ...query, query: '' } });
+  };
+
+  const onChangeWindowUnit = useCallback(
+    (selectedWindowUnit: string) => setParam('timeWindowUnit', selectedWindowUnit),
+    [setParam]
+  );
+
+  const onChangeWindowSize = useCallback(
+    (selectedWindowSize?: number) =>
+      selectedWindowSize && dispatch({ type: 'timeWindowSize', payload: selectedWindowSize }),
+    []
+  );
+
+  const onChangeSelectedThresholdComparator = useCallback(
+    (selectedThresholdComparator?: string) =>
+      setParam('thresholdComparator', selectedThresholdComparator),
+    [setParam]
+  );
+
+  const onChangeSelectedThreshold = useCallback(
+    (selectedThresholds?: number[]) =>
+      selectedThresholds && dispatch({ type: 'threshold', payload: selectedThresholds }),
+    []
+  );
+
+  const onChangeSizeValue = useCallback(
+    (updatedValue: number) => dispatch({ type: 'size', payload: updatedValue }),
+    []
+  );
+
+  return (
+    <Fragment>
+      <EuiTitle size="xs">
+        <h5>
+          <FormattedMessage
+            id="xpack.stackAlerts.searchThreshold.ui.conditionPrompt"
+            defaultMessage="When the number of documents match"
+          />
+        </h5>
+      </EuiTitle>
+
+      <EuiSpacer size="s" />
+
+      <DataViewSelectPopover
+        initialDataViewTitle={dataView.title}
+        initialDataViewId={dataView.id}
+        onSelectDataView={onSelectDataView}
+      />
+
+      <EuiSpacer size="s" />
+
+      <SearchBar
```

Comment:
@Dmitriynj it seems I can't switch to Lucene? 

https://user-images.githubusercontent.com/463851/168986971-1873dd98-597b-460b-b000-18268045d28d.mp4

Reply:
It seems once you've chosen a saved query, it can't be saved as new one 
![Bildschirmfoto 2022-05-18 um 10 08 20](https://user-images.githubusercontent.com/463851/168990258-8ba31837-321d-4981-8571-208b8c0397ba.png)

Reply:
KQL/Lucene works, thank you!

Reply:
Choosing of a saved query and saving it as new also works 👍

Reply:
saving a new saved query, doesn't select the new one:

https://user-images.githubusercontent.com/463851/169097001-173b6460-3ca1-465a-b1d6-9079628d01a4.mp4

Reply:
Fixed